File: CodeShare.Domain/Entities/CodeSnippet.cs
namespace CodeShare.Domain.Entities;

public class CodeSnippet
{
    public Guid Id { get; set; }

    public string? Title { get; set; }

    public string? Author { get; set; }

    public required string Code { get; set; }

    public DateTime CreatedAt { get; set; }
}

File: CodeShare.Client/src/app/models/codeSnippet.ts
export interface CodeSnippetBase {
  title?: string;
  author?: string;
  code: string;
}
export interface CodeSnippet extends CodeSnippetBase {
  id: string;
  createdAt: string;
  linesOfCode: number;
  html: string;
}
export interface CodeSnippetPreview extends CodeSnippetBase {
  linesOfCode: number;
  html: string;
}
